### DISCUSSION TITLE
Is it possible to show the page number of each title in the bookmark panel?

### DISCUSSION THREAD

**Discussion by Barrel-Titor** [sumatrapdfreader/sumatrapdf#3170] [Q&A]
What I mean is just like the screenshot, which comes from the online reader of [WeChat Reading](https://weread.qq.com/)
<img width="376" alt="Snipaste_2022-12-09_18-39-47" src="https://user-images.githubusercontent.com/14011748/206761492-c219fd02-a042-47f6-ab34-b1e9a5d13ac1.png">

I'm not sure whether this feature is realized or not, so I come here and ask for help. I've already searched in this forum but I didn't find similar questions (I'm sorry if it exists...)

**Comment by GitHubRulesOK** [comment]
Online readers are images and text
Not every document has a TOC (outline) this one has no page numbers for the chapters they need to be added (usually by the Author) just like bookmarks in SumatraPDF contents so not easy to add another column of data that takes time to build and may not be suited to all formats.
![image](https://user-images.githubusercontent.com/12754913/206793458-1f9523f1-8004-4f47-a918-810cd96ec743.png)

**Reply by Barrel-Titor** [reply]
Hi, thank you for your kind reply!
I've found in this link [How to display page number in ToC \ bookmarks panel](https://forum.sumatrapdfreader.org/t/how-to-display-page-number-in-toc-bookmarks-panel/4427) the exactly same question with mine, and it seems that the replier was you as well :)

> not easy to add another column of data that takes time to build and may not be suited to all formats

I'm not familiar with the coding behind pdf readers, so please forgive me if I get something wrong. In my simple cognition, since each bookmark must have a corresponding page number for jumping toward, and it can be easily shown like in the demo you gave in the link above, 
![sumatra fav](https://user-images.githubusercontent.com/14011748/206860641-f3f629e1-4e0e-4908-8c91-58480fe9ca64.png)
I think it will be a good idea to add another column and show the corresponding page number (if the auther has added TOC information). 

I tried the built-in reader on my mobile phone, and it can show the page numbers beside titles as well, so I'm guessing it is not a feature that is technically difficult to implement?
![duokan](https://user-images.githubusercontent.com/14011748/206860845-fe43d803-3807-471d-82f3-61fa18bd6a75.jpg)

I also tried several pdf readers on PC today, but I didn't find any one having this feature either, while I think it's useful when reading ebooks. Or maybe, is it because that developers will consider that such a feature would have a significant impact on time efficiency when opening very large files?

**Reply by Barrel-Titor** [reply]
Hi good morning! 
I've found in the code [sumatrapdf/src/TableOfContents.cpp](https://github.com/sumatrapdfreader/sumatrapdf/blob/master/src/TableOfContents.cpp) that there is (or maybe was) a flag `DISPLAY_TOC_PAGE_NUMBERS` (line 44) and it is used in `#ifdef` conditioning in line 46, 108, 749 and 858. 
So I guess this feature is already realized but not used in the latest versions. I wonder if it is possible to turn the flag on in the next few realeases, since this is really a great feature for me when reading ebooks.
